components/common-go/process/process.go

```diff
@@ -14,3 +21,36 @@ func IsNotChildProcess(err error) bool {
 
 	return (err.Error() == "wait: no child processes" || err.Error() == "waitid: no child processes")
 }
+
+var ErrForceKilled = errors.New("process didn't temrinate, so we send SIGKILL")
+
+// TerminateSync sends a SIGTERM to the given process and returns when the process has terminated but at most the given gracePeriod
+// if the gracePeriod ends we send SIGKILL to the process and return immediately with ErrForceKilled
+func TerminateSync[T any](pid int, gracePeriodOver <-chan T) error {
```

Comment:
Why do you use generics here?

I think if all you're looking for is a signal from the outside then the idiomatic would be `chan struct{}`

Reply:
It allows passing in `<- chan time.Time` (from time.After) but also other chans and communicate that we are only interested in the event.